README.md

```diff
@@ -5,14 +5,25 @@
 
 ## Routes
 
-### `GET /retrieval`
+### `PUT /retrieval`
+
+Start a new retrieval.
```

Comment:
When I see `PUT /retrieval`, I expect create-or-update semantics - the server should create a new resource or update an existing one, typically using some sort of an unique id.

Quoting from https://restfulapi.net/http-methods/#put

> Use PUT APIs primarily to update an existing resource (if the resource does not exist, then API may decide to create a new resource or not).
>
> If the request passes through a cache and the Request-URI identifies one or more currently cached entities, those entries SHOULD be treated as stale. Responses to PUT method are not cacheable.

IMO, this endpoint is creating (generating) a new retrieval job definition, therefore it's better to use the verb/method `POST`.

Quoting from https://restfulapi.net/http-methods/#post
> Use POST APIs to create new subordinate resources, e.g., a file is subordinate to a directory containing it or a row is subordinate to a database table.
> 
> When talking strictly about REST, POST methods are used to create a new resource into the collection of resources.
> 
> Responses to this method are not cacheable unless the response includes appropriate [Cache-Control](https://en.wikipedia.org/wiki/Web_cache#Cache_control) or [Expires](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html) header fields.
> 
> Please note that POST is neither safe nor idempotent, and invoking two identical POST requests will result in two different resources containing the same information (except resource ids).

I think we should also distinguish between retrieval job definition (what to retrieve) and retrieval report (whether we could download the content and telemetry data about the operation), and use different URLs (endpoint paths) for that. 

For example:

- `POST /retrieval-jobs` - create a new job
- `POST /retrieval-reports` - report a job outcome

This makes the API open to extensions, for example:

- `GET /retrieval-jobs?limit=10` - list 10 most recent retrieval jobs
- `GET /retrieval-jobs/123` - get details about the retrieval job with id 123
- `GET /retrieval-reports?limit=10` - list 10 most recent retrieval reports
- `GET /retrieval-reports/56` - get details about the retrieval report 56

I think we are not likely to implement these endpoints, I am listing them here to give more context to my suggestion.

Reply:
See comment below for discussion of the schema